tests/src/test/java/io/streamnative/pulsar/handlers/mqtt/ProxyTLSTest.java

```diff
@@ -89,4 +95,48 @@ public void testConnection(String topicName) throws Exception {
         received.ack();
         connection.disconnect();
     }
+
+    @Test(dataProvider = "mqttTopicNameAndFilter", timeOut = 20000, priority = 1)
+    @SneakyThrows
+    public void testSendAndConsumeWithFilter(String topic, String filter) {
+        MQTT mqtt0 = createMQTTProxyTlsClient();
+        // Set cert and keystore
+        File crtFile = new File(TLS_SERVER_CERT_FILE_PATH);
+        Certificate certificate = CertificateFactory
+                .getInstance("X.509").generateCertificate(new FileInputStream(crtFile));
+        KeyStore keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
+        keyStore.load(null, null);
+        keyStore.setCertificateEntry("server", certificate);
+        TrustManagerFactory trustManagerFactory = TrustManagerFactory
+                .getInstance(TrustManagerFactory.getDefaultAlgorithm());
+        trustManagerFactory.init(keyStore);
+        SSLContext sslContext = SSLContext.getInstance("TLS");
+        sslContext.init(null, trustManagerFactory.getTrustManagers(), null);
+        mqtt0.setSslContext(sslContext);
+        //
+        BlockingConnection connection0 = mqtt0.blockingConnection();
+        connection0.connect();
+        Topic[] topics = { new Topic(filter, QoS.AT_MOST_ONCE) };
+        String message = "Hello MQTT Proxy";
+        MQTT mqtt1 = createMQTTProxyTlsClient();
+        mqtt1.setSslContext(sslContext);
+        BlockingConnection connection1 = mqtt1.blockingConnection();
+        connection1.connect();
+        connection1.publish(topic, message.getBytes(), QoS.AT_MOST_ONCE, false);
+        // wait for the publish topic has been stored
+        Awaitility.await().untilAsserted(() -> {
+            CompletableFuture<List<String>> listOfTopics = pulsarServiceList.get(0).getNamespaceService()
+                    .getListOfTopics(NamespaceName.get("public/default"), CommandGetTopicsOfNamespace.Mode.PERSISTENT);
+            Assert.assertTrue(listOfTopics.join().size() >= 1);
+        });
+        connection0.subscribe(topics);
+        connection1.publish(topic, message.getBytes(), QoS.AT_MOST_ONCE, false);
```

Comment:
Why publish a message again?

Reply:
The first message publish to let the topic be stored in pulsar. And then consumers could subscribe to this topic.
 Because consumers subscribe to topics with wildcard have to find the namespace topics and then test if the wildcard matches the existed topics.